### PR TITLE
Multiple path support, bug fix, test suite naming

### PR DESCRIPTION
**Pull request by mnaughto**
I was originally going to submit an issue with a pull request to fix a bug where the first linting error message would be repeated for all remaining linting errors, but then I saw that you had removed the support for specifying directories.

I find the ability to specify paths for linting to be a nice feature, largely for convenience. I added support for a `files` option when initializing the module. Each path in the option is evaluated under a separate test.

I also added the ability to change the test suite name from "jshint", if desired.

Internally, I did a little refactoring. I think there are some edge cases that might be missed by the new version of the `run` function in index.js, but I haven't figured out the best way to handle them.

The README has also been updated to reflect the new features.

### CODE REVIEW

**Comment by ebdrup** [comment]
Hi could you explain how you are using the files. Originally I added the paths feature because I did not understand how to set up .jshintignore and .jshintrc files correctly myself. Once I understood how to set those up, I've not used the paths since.

Also your PR has some jshint errors ;-)

File: index.js [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
+			if (files.length === 0) {
+				//return done();
 			}
-		});
+			
+			if (opt.git) {
+				it('should pass for working directory', createTest(files));
+			} else {
+				files.forEach(function (file) {
+					it('should pass for ' + path.resolve(file), createTest([file]));
```
Comment (ebdrup) [inline index.js]: I would normally add a `JSON.stringify(path.resolve(file))` So any special chars show up nicely.

File: index.js [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
+}
+
 module.exports = function (opt) {
-	describe('jshint', function () {
-		it('should pass for working directory', function (done) {
-			this.timeout && this.timeout(30000);
-			if (opt && opt.git) {
-				return require('./git')(opt.git, run);
+	opt = opt || {};
+	describe(opt.title || 'jshint', function (done) {
```
Comment (ebdrup) [inline index.js]: this done can be removed it seems

File: index.js [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
-				}
-				return done();
+		return run(null, files);
+
+		function run(err, files) {
+			if (err) {
+				//return done(err);
+			}
+			if (files.length === 0) {
+				//return done();
```
Comment (ebdrup) [inline index.js]: you can remove this code that does nothing

File: index.js [hunk 4]
```diff
@@ -1,37 +1,58 @@
...
-		});
+			
+			if (opt.git) {
+				it('should pass for working directory', createTest(files));
+			} else {
+				files.forEach(function (file) {
+					it('should pass for ' + path.resolve(file), createTest([file]));
+				});
+			}	
+			//done();
```
Comment (ebdrup) [inline index.js]: I prefer not to have commented out code.

**Comment by mnaughto** [comment]
I have corrected the linting errors and I think errors occurring outside of tests are now handled correctly.

In regards to the `files` option, I just think it's a useful feature for making the tests more granular rather than having a single test for the entire code base. In my case, I'm passing three directories: one for back-end code, one for front-end code, and a third for some back-end utility code.
